lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -100,7 +100,11 @@ protected function onNotFoundMetadata($className)
     }
 
     /**
-     * {@inheritDoc}
+     * @param ClassMetadata      $class
+     * @param ?ClassMetadata     $parent
+     * @param bool               $rootEntityFound      True when there is another entity (non-mapped superclass) class above the current class in the PHP class hierarchy.
+     * @param list<class-string> $nonSuperclassParents All parent class names that are not marked as mapped superclasses, with the direct parent class being the first and the root entity class the last element.
+     * @psalm-suppress MoreSpecificImplementedParamType
```

Comment:
We're discussing `psalm-suppress` internally but haven't reached an agreement yet. Until then, let's stick to using `psalm.xml` for this please.

Reply:
I've tried that, but was unable to use `referencedMethod` in `psalm.xm` to pin it down to this particular method (only worked for the entire file). So I chose the annotation to be more precise... 

I don't mind, just tell me what you prefer.

Reply:
I have used the argument you are making in the internal discussion, but so far, not much feedback. I'd rather not unilaterally decide to switch to `@psalm-suppress` though, I need to get more maintainers to agree on it. Also see: https://github.com/vimeo/psalm/issues/9127

Reply:
> Much higher probability it will be cleaned up/removed once it is no longer necessary

In the internal thread, I mentioned another argument: the existence of `find-unused-psalm-suppress`. But issue handlers are catching up it seems: https://github.com/vimeo/psalm/issues/9102

Reply:
>  nobody is ever going to review them and clean up.

I guess this means I'm not a nobody :)

https://github.com/doctrine/orm/pull/10415